### ISSUE TITLE
Sphinx toctree warnings from included CHANGELOG.md

### ISSUE THREAD

**Issue by stevepiercy** [plone/volto#4422]
Ever since we started to include the `CHANGELOG.md` in the documentation, we had to resolve its path between both `plone/documentation` and `plone/volto`.

At first we attempted to use the Sphinx extension `ifconfig`, but we found that it generated misleading warnings that occasionally caught the attention of authors. `ifconfig` was never designed for this use case.

Next we removed `ifconfig` from Volto to silence the warnings in `plone/documentation`, but that resulted in dozens of warnings in Volto, like this:

`documentation/submodules/volto/docs/source/index.md:38: WARNING: toctree contains reference to document 'release-notes/index' that doesn't have a title: no link will be generated`

These warnings are just noise and prevent authors from spotting useful warnings.

@ksuess suggested creating a duplicate of `CHANGELOG.md` in `docs/`, and instead updating the reference there. That will remove the warnings.

However, as the generation of `CHANGELOG.md` is a step in the release of Volto, the release manager would need to agree that this would be a good step to automate, if possible. I can see that it could be useful for other repositories that include a change log in their docs.

@sneridagh @tiberiuichim @mauritsvanrees @ksuess @nileshgulia1 and any others, do you have thoughts about this proposal?

I would also like to update the comments in the `CHANGELOG.md` to update the URL to the docs and to add some clarity, per https://github.com/plone/documentation/issues/1431#issuecomment-1437514155

See also: https://github.com/plone/volto/pull/4404#issue-1588864779

**Comment by nileshgulia1** [comment]
@stevepiercy @ksuess Since we are updating the tips of volto submodule, wouldn't the changelog automatically get fetched when we are building the docs from "plone/documentation"?  I agree with https://github.com/plone/documentation/issues/1431#issue-1538209052 I don't mind "not seeing the changelog" when building Volto docs.

**Comment by stevepiercy** [comment]
The entire volto repository gets pulled in as a git submodule. Updating the tips just means pulling in the volto repo at that hash.

> Next we removed ifconfig from Volto to silence the warnings in plone/documentation, but that resulted in dozens of warnings in Volto, like this

IOW, that is the same as not including the changelog when building volto docs only. See https://github.com/plone/documentation/issues/1431#issuecomment-1434212761 

Another workaround would be to set the heading level of CHANGELOG.md one step lower throughout the page to avoid the warnings, and add a title to `release-notes/index.md`. I am not in favor of this option because it means having a duplicate title on the rendered page.

**Comment by stevepiercy** [comment]
What is the release workflow, or more to the point, after `CHANGELOG.md` is generated, only then can its links be tested. How do we manage that?

**Comment by davisagli** [comment]
@stevepiercy We run `release-it`, it updates the CHANGELOG from the towncrier entries and commits to the main branch. It's difficult to add a QA step here. Instead I'd suggest that we run link checks on the towncrier entries as they are added

**Comment by stevepiercy** [comment]
I think I can configure Sphinx to recognize specific file extensions and locations as MyST. I'll give that a shot, and report back. Thanks for the idea!

**Comment by stevepiercy** [comment]
@davisagli beyond the linkcheck, we still need a way to avoid other warnings that are generated due to `CHANGELOG.md` being outside the documentation root. Is there any way to add a step after `CHANGELOG.md` is generated to prepend the `html_meta` information (perhaps just insert it into the file now), then copy it to `docs/source/release-notes/index.md`?

**Comment by davisagli** [comment]
@stevepiercy release-it supports hooks that could probably be used to automate that as a step in the release process: https://github.com/release-it/release-it#hooks

We already use one of these to generate CHANGELOG.md using towncrier: https://github.com/plone/volto/blob/master/package.json#L188

**Comment by stevepiercy** [comment]
@davisagli something like this?

https://github.com/plone/volto/blob/63fbe9179d534b8b57d118fcb29e23a16510e271/package.json#L188

would become (line breaks added for legibility)

```json
      "after:bump": "pipx run towncrier build --draft --yes --version ${version} > .changelog.draft
&& pipx run towncrier build --yes --version ${version}
&& make corepackagebump VERSION=${version}
&& cp CHANGELOG.md docs/source/release-notes/index.md
&& git add docs/source/release-notes/index.md
&& git commit -m 'copy CHANGELOG.md'
&& git push", # or something similar to push it to the repo
```

I have a working demo where I manually copied `CHANGELOG.md` over, and the `meta_html` stuff has no impact on rendering docs. We should make sure it has no effect on releasing, too.

The only way I was able to get linkcheck to work for both `volto` and `documentation` repos was to create a symlink with a relative path. I added that as a step in both Makefiles for documentation and volto.

Here are new warnings, which may be safely ignored. We can consider suppressing them with `toc.excluded` but I like them as a reminder to look at the news folder for links, but more importantly sometimes developers add a file and forget to add it to a toctree or specifically exclude it. I added suppression for `WARNING: toctree contains reference to nonexisting document 'news*'` because this is a result of using a symlink with:

```python
suppress_warnings = [
    # "toc.excluded",
    "toc.not_readable",
]
```

## Volto

```
PATH/documentation/submodules/volto/docs/source/news/5131.documentation: WARNING: document isn't included in any toctree
PATH/documentation/submodules/volto/docs/source/news/5132.bugfix: WARNING: document isn't included in any toctree
```

PR coming in a bit.

**Comment by stevepiercy** [comment]
@davisagli @sneridagh could I get your feedback on modifying the command in this comment?

https://github.com/plone/volto/issues/4422#issuecomment-1694754052

I've got a proof of concept in https://github.com/plone/volto/pull/5135

**Comment by stevepiercy** [comment]
Another missing piece was https://github.com/plone/documentation/pull/1527 which I just merged. Let's run the gauntlet on docs again to see what happens, in both documentation and volto repos.

**Comment by stevepiercy** [comment]
Almost there. I have a couple of open PRs to resolve the last bits.

- https://github.com/plone/documentation/pull/1573
- https://github.com/plone/volto/pull/5375

When both of those are closed, then this issue may be closed.

**Comment by stevepiercy** [comment]
One more item needs to be merged before closing, the documentation of it being OK to ignore the toctree warnings, in https://github.com/plone/volto/pull/5377

**Comment by stevepiercy** [comment]
All the required issues and PRs have been resolved. Closing!
